### PR TITLE
Adding Node JS to OIE Embedded Authentication Guide

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Adding Node.js stack selector to OIE Embedded Authentication Guide
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-408181](https://oktainc.atlassian.net/browse/OKTA-408181)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 When multiple çonfigurations options are used simultaneously, the SDK chooses
 the option based on the following order:
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md:2]: the SDK -> the sample app

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
-* SDK Client constructor
+* SDK `config.js` file
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md:4]: The sample app uses configs from `config.js`, and the `config.js` loads information from environment variables (process.env). Devs can add env vars by either exporting from shell or using the `testenv` file.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/githubinstructions.md [hunk 3]
```diff
@@ -2,14 +2,14 @@ After you satisfy the software requirements, clone the
 [Okta Auth JavaScript SDK](https://github.com/okta/okta-auth-js)
 repo in a local directory on your computer.
 
-```bash
+```console
 git clone github https://github.com/okta/okta-auth-js.git
 cd okta-auth-js
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/githubinstructions.md:7]: as this guide is targeting to the sample app, how about `cd` to the sample app directory?

```
cd okta-auth-js/samples/generated/express-embedded-sign-in-widget
```

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/projectstructure.md [hunk 4]
```diff
@@ -3,7 +3,7 @@ The source code from the sample app and SDK are located here:
 `.../okta-auth-js/`. The project structure consists of the following:
 
 * `okta-auth-js/docs/idx.md`: Identity Engine introduction and reference
-* `okta-auth-js/samples`: Sample directory
+* `okta-auth-js/samples`: Samples directory
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/projectstructure.md:6]: okta-auth-js/samples -> okta-auth-js/samples/generated

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md [hunk 5]
```diff
@@ -1,41 +1,12 @@
-## TO DO for NodeJS
 
-#### Step 1: Set up your app for .Net 4.8 or greater
+#### Step 1: Install the SDK into your project application
 
-The SDK and samples are built using .Net 4.8.
+Download and install the SDK into your Express application. See [Building the SDK](https://github.com/okta/okta-auth-js#building-the-sdk) for further information.
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md:4]: The build step won't be needed if the devs are using the SDK in their own app. 

The steps should be:
Firstly, install okta-auth-js, `npm install @okta/okta-auth-js`
Then explain how to use the SDK methods. Details can be found [here](https://github.com/okta/okta-auth-js/blob/master/docs/idx.md)

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md [hunk 6]
```diff
@@ -1,41 +1,12 @@
...
 
-#### Step 2: Add the Okta SDK Nuget Packages
+#### Step 2: Install the module dependencies
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md:6]: step 1 & 2 can be combined into one step

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md [hunk 7]
```diff
@@ -1,41 +1,12 @@
...
-
-See the following sample code for more details:
-
-```csharp
-using Okta.Idx.Sdk;
-using Okta.Sdk.Abstractions;
-
-var idxAuthClient = new IdxClient();
+```console
+npm install
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md:11]: not need, as it's the devs own app, they should have control of the app's dependencies, we probably only need to expose the sdk installation detail.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md [hunk 8]
```diff
@@ -1,3 +1,3 @@
 
-* Node.js v12.13.0 or above (do we have a minimum version?)
-* Express Framework 14.17.1 or above
+* Node.js as defined by the `.npmrc ` version.
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md:2]: let's be specific here, Node.js version >= 12.0.0

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md [hunk 9]
```diff
@@ -1,3 +1,3 @@
...
+* Installed module dependencies from the `package.json` file. See [Install module dependencies](/docs/guides/oie-embedded-common-download-setup-app/nodejs/main/#install-module-dependencies).
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md:3]: This part is covered in `Install module dependencies` section

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
+## Steps to test the sample app
+
+1. If not already done, set up your Okta org by completing these steps: [Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:3]: nit: looks like all steps are with `1.`

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+1. If not already done,
+   [download and set up the sample app](/docs/guides/oie-embedded-common-download-setup-app/nodejs/main/).
+1. Locate the sample apps solution file in the following path:
+`...\okta-auth-js\samples\generated\express-embedded-auth-with-sdk`
+1. Open `express-embedded-auth-with-sdk` using Visual Studio or the editor of your choice.
+1. Add a `testenv` configuration file. See [Option 1: Configuration file](/docs/guides/oie-embedded-common-download-setup-app/nodejs/main/#option-1-configuration-file) for further information.
+1. From the terminal, navigate to the `express-embedded-auth-with-sdk` sample directory and run the application: `npm run start`. This process sets the environment variables and starts the web server. Check the console to see that the environment variables are set.
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:10]: `npm install` is also needed here

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
+## Steps to test the sample app
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:1]: maybe `test` -> `run` ?

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+1. If not already done,
+   [download and set up the sample app](/docs/guides/oie-embedded-common-download-setup-app/nodejs/main/).
+1. Locate the sample apps solution file in the following path:
+`...\okta-auth-js\samples\generated\express-embedded-auth-with-sdk`
+1. Open `express-embedded-auth-with-sdk` using Visual Studio or the editor of your choice.
+1. Add a `testenv` configuration file. See [Option 1: Configuration file](/docs/guides/oie-embedded-common-download-setup-app/nodejs/main/#option-1-configuration-file) for further information.
+1. From the terminal, navigate to the `express-embedded-auth-with-sdk` sample directory and run the application: `npm run start`. This process sets the environment variables and starts the web server. Check the console to see that the environment variables are set.
+1. Add an `okta.yaml` configuration file. See [Option 1: YAML configuration file](/docs/guides/oie-embedded-common-download-setup-app/aspnet/main/#option-1-configuration-file) for more information on how to configure and where to place the configuration file.
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:11]: is the yaml file from the .net sample? it's not needed in nodejs sample

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/nodejs/integrationsteps.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
+## Integration steps
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/nodejs/integrationsteps.md:1]: Looks like it's still targeting to the .net sample.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-load/nodejs/integrationsteps.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+      </script>
+
+</div>
+```
+
+The `WidgetConfig` is referenced in the `login.js` file, which uses values defined in the `config.js` file:
+
+```JavaScript
+console.log('renderLoginWithWidget: using interaction handle: ', interactionHandle);
+      const { clientId, redirectUri, issuer, scopes } = getConfig().webServer.oidc;
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-load/nodejs/integrationsteps.md:45]: nit: indent

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-load/nodejs/integrationsteps.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+        codeChallenge,
+        codeChallengeMethod,
+      };
+```
+
+The Okta Sign-In Widget renders in the sign-in page when the `router.get('/login', (req, res, next) =>` call is triggered.
+
+```JavaScript
+res.render('login', {
+        siwVersion: '{{siwVersion}}',
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-load/nodejs/integrationsteps.md:66]: nit: indent

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
+## Integration steps
```
Comment (shuowu-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md:1]: code snippets are still referring to the .net sample

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md (hunk 2 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/configorder.md:4]: Thanks, added this info.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md (hunk 5 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md:4]: Thanks, I recast this page with these updates.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md (hunk 6 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/sdkforyourapp.md:6]: As mentioned, I recast this section.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md (hunk 9 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-common-download-setup-app/main/nodejs/softwarerequirements.md:3]: Yes, I agree. A bit of overkill. Just trying to stick with our template, but I can remove.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md (hunk 10 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:3]: Thanks, I believe this renders in order with Vue.Press but will confirm.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md (hunk 13 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:11]: Yes, sorry, used the .net content as a base. Thanks, will clean up.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/nodejs/integrationsteps.md (hunk 14 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-basic-sign-in/nodejs/integrationsteps.md:1]: Yes, recently updated in a later commit.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md (hunk 17 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/guides/oie-embedded-widget-use-cases/oie-embedded-widget-use-case-sign-in-soc-idp/nodejs/integrationsteps.md:1]: Yes, I need to work on this one this afternoon.

File: packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md [hunk 18]
```diff
@@ -0,0 +1,42 @@
+## Steps to test the sample app
+
+1. If not already done, set up your Okta org by completing these steps: [Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/aspnet/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-run-sample/main/nodejs/testapp.md:3]:
```suggestion
1. If not already done, set up your Okta org by completing these steps: [Set up your Okta org (for password factor only use cases)](/docs/guides/oie-embedded-common-org-setup/nodejs/main/#set-up-your-okta-org-for-password-factor-only-use-cases).
```

**Review by vanngo-okta — APPROVED** [review]
LGTM (small nits) 🚢

**Pull request merged**